File: gitlfs-client/src/main/java/ru/bozaro/gitlfs/client/io/UrlStreamProvider.java
package ru.bozaro.gitlfs.client.io;

import javax.annotation.Nonnull;
import java.io.IOException;
import java.io.InputStream;
import java.net.URL;

/**
 * Create stream by URL.
 *
 * @author Artem V. Navrotskiy
 */
public class UrlStreamProvider implements StreamProvider {
  @Nonnull
  private final URL url;

  public UrlStreamProvider(@Nonnull URL url) {
    this.url = url;
  }

  @Nonnull
  @Override
  public InputStream getStream() throws IOException {
    return url.openStream();
  }
}

File: gitlfs-client/src/main/kotlin/ru/bozaro/gitlfs/client/io/UrlStreamProvider.kt
package ru.bozaro.gitlfs.client.io

import java.io.IOException
import java.io.InputStream
import java.net.URL

/**
 * Create stream by URL.
 *
 * @author Artem V. Navrotskiy
 */
class UrlStreamProvider(private val url: URL) : StreamProvider {
    @get:Throws(IOException::class)
    override val stream: InputStream
        get() = url.openStream()
}
